rust/numaflow-throttling/src/lib.rs

```diff
@@ -42,7 +46,7 @@ pub struct RateLimit<W> {
     /// Max number of tokens ever filled till now. The next refill relies on the current max allowed
     /// value. For the first run this will be set to `burst` and will be slowly incremented to the
     /// `max` value.
-    max_ever_filled: Arc<AtomicUsize>,
+    max_ever_filled: Arc<Mutex<f32>>,
```

Comment:
let's use `usize` and ciel to make sure it will be `usize`, do we care for `f32`. It has to be an absolute value.

Reply:
`ceil` doesn't work because if the slope is 0.1 and ram up duration is 10s, the next token should be available at 10th second. If we use ceil it will be available in the 2nd second.